### PR TITLE
[1.10.x] xds: allow only one outstanding delta request at a time

### PR DESCRIPTION
**Pull request by rboyer**
Backport of #12236 to 1.10.x

### CODE REVIEW

File: agent/xds/delta.go [hunk 1]
```diff
@@ -309,13 +305,7 @@ func (s *Server) processDelta(stream ADSDeltaStream, reqCh <-chan *envoy_discove
 						op.TypeUrl, err)
 				}
 				if sent {
-					sentType[op.TypeUrl] = struct{}{}
-					if generator.ProxyFeatures.IncrementalXDSUpdatesMustBeSerial {
```
Comment (rboyer) [inline agent/xds/delta.go:313]: Basically just inlined this older feature flag where it was actually right.

**Review by dhiaayachi — APPROVED** [review]
🚀

**Review by eculver — APPROVED** [review]
I have very little context on this, but I trust the logic around the flags and the testing so 👍🏼

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/578102.

**Pull request merged**